source/extensions/filters/http/ext_proc/processor_state.h

```diff
@@ -30,6 +30,24 @@ class QueuedChunk {
 };
 using QueuedChunkPtr = std::unique_ptr<QueuedChunk>;
 
+class ChunkQueue {
+public:
+  ChunkQueue() = default;
+  ChunkQueue(const ChunkQueue&) = delete;
+  ChunkQueue& operator=(const ChunkQueue&) = delete;
+  uint32_t bytesEnqueued() const { return bytes_enqueued_; }
+  bool empty() const { return queue_.empty(); }
+  void push(Buffer::Instance& data, bool end_stream, bool delivered);
+  absl::optional<QueuedChunkPtr> pop(bool undelivered_only);
+  const QueuedChunk& consolidate(bool delivered);
+
+private:
+  // A queue of chunks that were sent in streaming mode
```

Comment:
Is this just used in streaming mode? How does this comment work with the idea that each entry is labeled `delivered` or not?

Reply:
Clarifying the comments.